rfc/rfc_001_stamping-txs.md

```diff
@@ -0,0 +1,117 @@
+                             |                                      |
+                             |                                      |
+                             |          Tx                          |
+                             |         +-------------------+\       |
+                             \-------->| STAMP             ||       |
+                                       +-------------------+|       |
+                                       | PAYLOAD           ||       |
+                                       |                   ||-------/
+                                       |                   ||
+                                       |                   ||
+                                       +-------------------+|
+                                       | SIGNATURE         ||
+                                       +-------------------+/
+```
+
+### Where stamped transactions will be saved?
+
+Full nodes can have separate database for saving stamped transactions.
+This database will only be used for querying transaction. A node can choose to save all txs, part of them or nothing.
+Nodes can query transaction by hash from each other. Users also can save their own transactions in their own wallets.
+
+
+### Can a stamped transaction be lost?
+
+Probably yes and it's not a curse. Should we care about transaction happened 10 years ago and try to keep it in our DB?
+By decoupling transaction from the block, transactions will have their own way to survive.
+
+
+## Decision
+
+## Status
+
+## Consequences
+
+* Reducing block size.
+
+  Stamping transaction can reduce the block size dramatically. For keeping a stamped transaction inside a block we need to store only 32 bytes hash of transaction instead of whole transaction data. In the bitcoin, the average transaction size has increased from 450 bytes in Jan 2013 to almost 600 bytes in Oct 2015. In the below chart you can see how stamped transaction could reduce the size of bitcoin blockchian if they could had applied from the beginning:
+
+  ![Bitcoin average block size](img/btc-block-size.png)
+  ![Bitcoin blockchain size](img/btc-blockchain-size.png)
+
+* Easy light clint implementation for PoS
+
+  Implementing a light client for PoW is easy since having block header is fair enough to check validity of a transaction using merkle tree.
+  However in PoS implementing a light client is a big pain. By stamping transaction light client implementation will be easy. A light client doesn't need to store any transaction data.
+
+
+* Mitigating any possible forks.
+
+  *Nothing at stake attack* and *long range attack* are most fundamental attacks in PoS blockchains. There is no solution to overcome long range attack in PoS   blockchain. Stamping transactions can mitigate this issue by binding transactions to the main fork. A stamped transaction is always loyal to the main fork. Byzantine forks aren't able to put them in their fork because the stamp is not belongs to their fork.
```

Comment:
I think this is the biggest advantage of stamping transactions.

Reply:
Yap. Actually I wanted to find a way for solving "Long Range attack", however it also reduce the size of block, and it's beautiful. In Tendermint block structure is a bit complex. I think `block_id` can be simply the block hash. No need to have block parts. It took time for me to understand what is block part and why we need it. I think that is avoidable.

Reply:
I still think this is cheating for reducing size, similar to how segwit is really just cheating by moving the goal post and not actually solving anything. Using validity proofs over the entire chain (such as Coda) is an example of solving this without cheating.